pkg/store/flag.go

```diff
@@ -0,0 +1,74 @@
+package store
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+// TimeOrDurationValue is a custom kingping parser for time in RFC3339
+// or duration in Go's duration format, such as "300ms", "-1.5h" or "2h45m".
+// Only one will be set.
+type TimeOrDurationValue struct {
+	t   *time.Time
+	dur *model.Duration
+}
+
+// Set converts string to TimeOrDurationValue
+func (tdv *TimeOrDurationValue) Set(s string) error {
+	t, err := time.Parse(time.RFC3339, s)
+	if err != nil {
```

Comment:
please keep bigger code blocks with smaller indent. This means filtering err == nil first and then handling the err case

Reply:
nice catch :heart: